File: common/src/main/java/net/cliff3/maven/common/util/crypto/AESCrypto.java
package net.cliff3.maven.common.util.crypto;

import lombok.Getter;

/**
 * AES 암호화 결과
 *
 * @author JoonHo Son
 * @version 1.0.0 2019-12-13 최초 작성
 * @since 1.0.0
 */
public class AESCrypto {
    /**
     * Salt
     */
    @Getter
    private byte[] salt;

    /**
     * 암호화된 결과
     */
    @Getter
    private byte[] encrypted;

    /**
     * Initialize vector
     */
    @Getter
    private byte[] iv;

    /**
     * Default constructor
     *
     * @param salt      {@link #salt}
     * @param encrypted {@link #encrypted}
     * @param iv        {@link #iv}
     */
    public AESCrypto(byte[] salt, byte[] encrypted, byte[] iv) {
        this.salt = salt;
        this.encrypted = encrypted;
        this.iv = iv;
    }
}

File: common/src/main/java/net/cliff3/maven/common/util/crypto/AESCrypto.kt
package net.cliff3.maven.common.util.crypto

/**
 * AES 암호화 결과.
 *
 * @author JoonHo Son
 * @version 1.0.0 2019-09-09 최초 작성
 * @since 1.0.0
 */
class AESCrypto(salt: ByteArray, encrypted: ByteArray, iv: ByteArray) {
    /**
     * Salt
     */
    var salt: ByteArray = salt
        private set

    /**
     * 암호화된 결과.
     */
    var encrypted: ByteArray = encrypted
        private set

    /**
     * Initialize vector
     */
    var iv: ByteArray = iv
        private set
}